library/core/src/main/java/com/google/android/exoplayer2/source/UnreportedDiscontinuityException.java

```diff
@@ -0,0 +1,26 @@
+package com.google.android.exoplayer2.source;
+
+import android.net.Uri;
+import com.google.android.exoplayer2.C;
+
+/**
+ * Thrown from the loader thread when an attempt is made to commit a sample that is far
+ * deviant from the expected sequence of timestamps in the SampleStream.   This is likely
+ * caused by a discontinuity in a segment that was not split and reported by metadata in
+ * an HLS (EXT-X-DISCONTINUITY) or DASH stream.
+ */
+public class UnreportedDiscontinuityException extends RuntimeException {
```

Comment:
IMO make it extend `IOException` so you don't have to catch and re-throw.

The naming is also a bit confusing, because the exception is by definition reporting the discontinuity :). Maybe call it `UnexpectedDiscontinuityException`.

Also make it final.